File: languages/div.cpp
#include <iostream>
#include <stdexcept>

class DivError : public std::runtime_error {
public:
    DivError() : std::runtime_error("division by zero") {}
};

double div(double x, double y) {
    if (y == 0.0) {
        throw DivError();
    }
    return x / y;
}

int main() {
    double x = 1.0;
    double y = 0.0;
    
    try {
        double z = div(x, y);
        std::cout << "The result is: " << z << std::endl;
    } catch (const DivError& e) {
        std::cerr << "Division failed: " << e.what() << std::endl;
    }
    
    return 0;
}

File: languages/div.php
<?php

class DivError extends Exception {
    public function __construct($message = "division by zero", $code = 0, Throwable $previous = null) {
        parent::__construct($message, $code, $previous);
    }
}

function div($x, $y) {
    if ($y == 0.0) {
        throw new DivError();
    }
    return $x / $y;
}

function main() {
    $x = 1.0;
    $y = 0.0;
    
    try {
        $z = div($x, $y);
        echo "The result is: $z\n";
    } catch (DivError $e) {
        fwrite(STDERR, "Division failed: " . $e->getMessage() . "\n");
    }
}

main();

?>